pdns/bpf-filter.cc

```diff
@@ -320,6 +321,23 @@ BPFFilter::BPFFilter(std::unordered_map<std::string, MapConfiguration>& configs,
     throw std::runtime_error("Unsupported eBPF map format, the current internal implemenation only supports the legacy format");
   }
 
+  struct rlimit old_limit;
+
+  if ( getrlimit(RLIMIT_MEMLOCK, &old_limit) != 0) {
+    throw std::runtime_error("Unable to get memory lock limit: " + stringerror());
+  }
+
+  /* Check if the current soft memlock limit is 64k */ 
+  if (old_limit.rlim_cur < 67108865) {
+    struct rlimit new_limit;
+    new_limit.rlim_cur = 1073741824; /* Increase soft limit to 1024k */
+    new_limit.rlim_max = 1073741824; /* Increase hard limit to 1024k */
+  
+    if (setrlimit(RLIMIT_MEMLOCK, &new_limit) != 0) {
+      throw std::runtime_error("Unable to increase memory lock limit: " + stringerror());
```

Comment:
I think it might be better not to throw here, as in some cases (very small map, or different kernel version that uses less locked memory for example) we might be able to start even though we could not raise the memory. So something like:
```c++
errlog("Unable to raise the maximum amount of locked memory for eBPF from %d to %d, consider raising RLIMIT_MEMLOCK or setting LimitMEMLOCK=infinity in the systemd unit: %s", old_limit.rlim_cur, new_limit.rlim_cur, stringerror());
```